src/Compilers/Core/Portable/InternalUtilities/ConcurrentBag.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.InternalUtilities
+{
+    internal class ConcurrentBag<T>
+        : System.Collections.Concurrent.ConcurrentBag<T>, ICollection<T>
```

Comment:
This seems like an implementation for the Compiler classes rather than a general replacement for `ConcurrentBag<T>`. Consider making this a nested class in `CommonCompiler`, perhaps with a more specific name.